### PR TITLE
Explain how to provide a stack trace

### PR DESCRIPTION
**Pull request by greg0ire**
I miss a reference explaining how to do that properly and keep asking people to do it, so maybe here is the right place?

### CODE REVIEW

**Comment by greg0ire** [comment]
As a bonus, I added a link to the Slack channel.

**Review by Simperfit — APPROVED** [review]

File: contributing/code/bugs.rst [hunk 1]
```diff
@@ -14,8 +14,9 @@ Before submitting a bug:
 * Double-check the official :doc:`documentation </index>` to see if you're not misusing the
   framework;
 
-* Ask for assistance on `Stack Overflow`_ or on the #symfony `IRC channel`_
-  if you're not sure if your issue is really a bug.
+* Ask for assistance on `Stack Overflow`_ on the #symfony `IRC channel`_, or on
+  the #support channel of `the Symfony Slack` if you're not sure if your issue
```
Comment (Simperfit) [inline contributing/code/bugs.rst]: Maybe we could add the link of the[ symfony slack](https://symfony.com/slack-invite)

Comment (PeterFour) [inline contributing/code/bugs.rst]: I think he markup is broken for the Slack link

Comment (greg0ire) [inline contributing/code/bugs.rst]: You're right, let's fix it.

File: contributing/code/bugs.rst [hunk 2]
```diff
@@ -32,9 +33,16 @@ If your problem definitely looks like a bug, report it using the official bug
 * Give as much detail as possible about your environment (OS, PHP version,
   Symfony version, enabled extensions, ...);
 
+* If you want to provide a stack trace, be sure to provide the plain text
+  version, which should appear at the bottom of the page. *Do not* provide it
+  as a screenshot, since search engines will not be able to index the text
+  inside them. If the stack trace is long, consider enclosing it in a
+  ``<details>`` html tag.
```
Comment (Simperfit) [inline contributing/code/bugs.rst]: Maybe we should warn people about security on what they put on the `<details>`

Comment (PeterFour) [inline contributing/code/bugs.rst]: I think this even stand for SO too

**Review by Simperfit — APPROVED** [review]

File: contributing/code/bugs.rst (hunk 2 above)
Comment (greg0ire) [inline contributing/code/bugs.rst]: Yeah but the `<details>` part does not, and SO has its own reference.

Comment (greg0ire) [inline contributing/code/bugs.rst]: Maybe IRC and Slack need their own reference?

Comment (PeterFour) [inline contributing/code/bugs.rst]: Correct, I meant that in the general sense, pasting stack trace somewhere need to be done carefully

File: contributing/code/bugs.rst [hunk 3]
```diff
@@ -32,9 +33,19 @@ If your problem definitely looks like a bug, report it using the official bug
 * Give as much detail as possible about your environment (OS, PHP version,
   Symfony version, enabled extensions, ...);
 
+* If you want to provide a stack trace you got on an html page, be sure to
```
Comment (javiereguiluz) [inline contributing/code/bugs.rst]: I think I'd recommend the opposite: always share the stack traces as images because they offer some protection against Google, etc. Remember that traces can contain sensitive information, so we don't make them searchable on Internet.

Comment (greg0ire) [inline contributing/code/bugs.rst]: IMO sensitive data is the exception, not the rule, and a big drawback of using them that I did not mention is that you can't copy paste from them, making the work harder for people who want to check files mentioned in it. Also, these images can be resized and illegible. Also if not cropped properly, they might contain even more sensitive data, like what the desktop of the reporter looks like. Finally, even with a screenshot, you should redact sensitive information, so...

File: contributing/code/bugs.rst [hunk 4]
```diff
@@ -32,9 +33,19 @@ If your problem definitely looks like a bug, report it using the official bug
...
+  provide the plain text version, which should appear at the bottom of the
+  page. *Do not* provide it as a screenshot, since search engines will not be
+  able to index the text inside them. Same goes for errors encountered in a
+  terminal, do not take a screenshot, but learn how to copy/paste from it. If
+  the stack trace is long, consider enclosing it in a ``<details>`` html tag.
+  Be wary that stack traces may contain sensible information, and if it is the
```
Comment (sstok) [inline contributing/code/bugs.rst]: This should be made bold to make it more important (end less change to miss it).

File: contributing/code/bugs.rst (hunk 3 above)
Comment (sstok) [inline contributing/code/bugs.rst]: If you expose sensitive information you are already F**ked, posting this in a screenshot will not prevent anyone from abusing this information. Not to mention that search engines can perform OCR.

Comment (Pierstoval) [inline contributing/code/bugs.rst]: I'm also for plain text, because it allows search engines to match exception messages that people encounter, so they may be redirected to the correct issue just by googling the exception itself

File: contributing/code/bugs.rst [hunk 5]
```diff
@@ -32,9 +33,19 @@ If your problem definitely looks like a bug, report it using the official bug
 * Give as much detail as possible about your environment (OS, PHP version,
   Symfony version, enabled extensions, ...);
 
+* If you want to provide a stack trace you got on an html page, be sure to
+  provide the plain text version, which should appear at the bottom of the
+  page. *Do not* provide it as a screenshot, since search engines will not be
+  able to index the text inside them. Same goes for errors encountered in a
+  terminal, do not take a screenshot, but learn how to copy/paste from it. If
+  the stack trace is long, consider enclosing it in a ``<details>`` html tag.
```
Comment (Pierstoval) [inline contributing/code/bugs.rst]: Maybe a link to [a documentation example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details)?

File: contributing/code/bugs.rst [hunk 6]
```diff
@@ -32,9 +33,19 @@ If your problem definitely looks like a bug, report it using the official bug
...
+  **Be wary that stack traces may contain sensitive information, and if it is
+  the case, be sure to redact them prior to posting your stack trace.**
```
Comment (Pierstoval) [inline contributing/code/bugs.rst]: `s/redact/remove/`

File: contributing/code/bugs.rst (hunk 5 above)
Comment (greg0ire) [inline contributing/code/bugs.rst]: I'll try, let's hope the backticks I used do not confuse rst

File: contributing/code/bugs.rst (hunk 6 above)
Comment (greg0ire) [inline contributing/code/bugs.rst]: What's wrong with "redact"? IMO it's very appropriate here.

File: contributing/code/bugs.rst [hunk 7]
```diff
@@ -32,9 +33,20 @@ If your problem definitely looks like a bug, report it using the official bug
 * Give as much detail as possible about your environment (OS, PHP version,
   Symfony version, enabled extensions, ...);
 
+* If you want to provide a stack trace you got on an html page, be sure to
+  provide the plain text version, which should appear at the bottom of the
+  page. *Do not* provide it as a screenshot, since search engines will not be
+  able to index the text inside them. Same goes for errors encountered in a
+  terminal, do not take a screenshot, but learn how to copy/paste from it. If
+  the stack trace is long, consider enclosing it in a ` ``<details>`` html tag`_.
```
Comment (greg0ire) [inline contributing/code/bugs.rst]: This link doesn't work, if anyone know how to fix it I'm all ears

File: contributing/code/bugs.rst [hunk 8]
```diff
@@ -32,9 +33,20 @@ If your problem definitely looks like a bug, report it using the official bug
...
   Symfony version, enabled extensions, ...);
 
+* If you want to provide a stack trace you got on an html page, be sure to
+  provide the plain text version, which should appear at the bottom of the
+  page. *Do not* provide it as a screenshot, since search engines will not be
+  able to index the text inside them. Same goes for errors encountered in a
+  terminal, do not take a screenshot, but learn how to copy/paste from it. If
+  the stack trace is long, consider enclosing it in a `<details> html tag`_.
+  **Be wary that stack traces may contain sensitive information, and if it is
+  the case, be sure to redact them prior to posting your stack trace.**
```
Comment (alexislefebvre) [inline contributing/code/bugs.rst]: Did you mean _remove_ instead of _redact_?

Comment (greg0ire) [inline contributing/code/bugs.rst]: No, I did not but you are the second person to ask me that, so let me check the meaning.

File: contributing/code/bugs.rst (hunk 6 above)
Comment (greg0ire) [inline contributing/code/bugs.rst]: cc @alexislefebvre

File: contributing/code/bugs.rst (hunk 8 above)
Comment (greg0ire) [inline contributing/code/bugs.rst]: @alexislefebvre @Pierstoval it really seems appropriate : http://www.linguee.fr/francais-anglais/search?source=anglais&query=redact WDYT?

Comment (greg0ire) [inline contributing/code/bugs.rst]: I see both of you are french so maybe you're just unaware that redact is a *faux-ami* ?

Comment (alexislefebvre) [inline contributing/code/bugs.rst]: You're right, I didn't know the meaning of *redact*. :sweat_smile: Maybe you should simply write *edit*?

Comment (Pierstoval) [inline contributing/code/bugs.rst]: I did know the meaning of "redact" so it didn't shocked me 🤣 

Even though, if some people are confused, maybe the word should be changed without breaking the global understanding of the paragraph. Not every Symfony user is english-native 🙂

Comment (greg0ire) [inline contributing/code/bugs.rst]: @alexislefebvre *edit* does not convey the full meaning, and I don't think we should dumb it down with remove. I think people can easily infer the meaning of the word if they don't know it already, and that means they will have learned something, so IMO it's better to stick with that.

Comment (Pierstoval) [inline contributing/code/bugs.rst]: 👍 then

**Review by xabbuh — APPROVED** [review]
I really like this a lot. 👍

**Comment by greg0ire** [comment]
Thanks @xabbuh :blush:

**Review by Pierstoval — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]

**Comment by xabbuh** [comment]
Thank you @greg0ire.
